### PR TITLE
🐛 Fixes catalog's synchronization background task continues errors due to faulty service info

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Prevents the background sync task in `catalog` from continuously "stopping and restarting" due to errors in the director's response for certain items in the list. This change ensures that the task can still complete for the remaining items despite errors.

This is the error recurrently logged by the background sync task:
```
log_level=DEBUG | log_timestamp=2024-09-10 13:40:01,893 | log_source=httpcore.http11:atrace(85) | log_uid=None | log_msg=response_closed.complete
log_level=ERROR | log_timestamp=2024-09-10 13:40:01,893 | log_source=simcore_service_catalog.core.background_tasks:_sync_services_task(190) | log_uid=None | log_msg=Unexpected error while syncing registry entries, restarting now...
Traceback (most recent call last):
  File "/home/scu/.venv/lib/python3.11/site-packages/simcore_service_catalog/core/background_tasks.py", line 177, in _sync_services_task
    await _run_sync_services(app)
  File "/home/scu/.venv/lib/python3.11/site-packages/simcore_service_catalog/core/background_tasks.py", line 165, in _run_sync_services
    await _ensure_registry_and_database_are_synced(app)
  File "/home/scu/.venv/lib/python3.11/site-packages/simcore_service_catalog/core/background_tasks.py", line 114, in _ensure_registry_and_database_are_synced
    await _create_services_in_database(
  File "/home/scu/.venv/lib/python3.11/site-packages/simcore_service_catalog/core/background_tasks.py", line 74, in _create_services_in_database
    ) = await access_rights.evaluate_default_policy(app, service_metadata)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scu/.venv/lib/python3.11/site-packages/simcore_service_catalog/services/access_rights.py", line 73, in evaluate_default_policy
    if _is_frontend_service(service) or await _is_old_service(app, service):
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scu/.venv/lib/python3.11/site-packages/simcore_service_catalog/services/access_rights.py", line 43, in _is_old_service
    await client.get(
  File "/home/scu/.venv/lib/python3.11/site-packages/simcore_service_catalog/services/director.py", line 94, in request_wrapper
    return _unenvelope_or_raise_error(resp)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scu/.venv/lib/python3.11/site-packages/simcore_service_catalog/services/director.py", line 71, in _unenvelope_or_raise_error
    raise HTTPException(resp.status_code, detail=msg)
fastapi.exceptions.HTTPException: 404
```
We have improved the resilience of these routines by catching expected exceptions, issuing a warning, and continuing to process the remaining elements in the loop without interruption.

#### Misc
- extra tmp timeout for @mguidon until issue is fully resolved

## Related issue/s

- fixes https://github.com/ITISFoundation/osparc-simcore/issues/6318
- part of https://github.com/ITISFoundation/osparc-simcore/issues/6201
- resolves part of https://github.com/ITISFoundation/osparc-simcore/issues/1854

## How to test

```
cd services/catalog
make install-dev
# when parameter `director_fails=True`
pytest -vv tests/unit/with_dbs -k test_registry_sync_task
```
## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/catalog/src/simcore_service_catalog/core/background_tasks.py [hunk 1]
```diff
@@ -62,32 +64,45 @@ def _by_version(t: tuple[ServiceKey, ServiceVersion]) -> Version:
...
-            service_access_rights,
-        )
+            # set the service in the DB
+            await services_repo.create_or_update_service(
+                ServiceMetaDataAtDB(**service_metadata.dict(), owner=owner_gid),
+                service_access_rights,
+            )
+
+        except (HTTPException, ValidationError, SQLAlchemyError) as err:
+            # Resilient to single failures: errors in individual (service,key) should not affect the wholse serices
```
Comment (matusdrobuliak66) [inline services/catalog/src/simcore_service_catalog/core/background_tasks.py:97]: typo: should not affect the whole service.

**Review by sanderegg — APPROVED** [review]
perfect, thanks!

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/catalog/services.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ async def _call(
             user_id=user_id,
             limit=limit,
             offset=offset,
-            timeout_s=20 * RPC_REQUEST_DEFAULT_TIMEOUT_S,
+            timeout_s=40 * RPC_REQUEST_DEFAULT_TIMEOUT_S,
```
Comment (GitHK) [inline packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/catalog/services.py:57]: Q: I don't get why you have to multiply here the value, can't we just use one single constant, which is simpler to read and comprehend?

File: services/catalog/src/simcore_service_catalog/core/background_tasks.py (hunk 1 above)
Comment (GitHK) [inline services/catalog/src/simcore_service_catalog/core/background_tasks.py:97]:
```suggestion
            # Resilient to single failures: errors in individual (service,key) should not affect the all the services
```

Comment (GitHK) [reply services/catalog/src/simcore_service_catalog/core/background_tasks.py:97]: or something like this?

File: services/catalog/src/simcore_service_catalog/core/background_tasks.py [hunk 3]
```diff
@@ -62,32 +64,45 @@ def _by_version(t: tuple[ServiceKey, ServiceVersion]) -> Version:
...
+            # and stop the background task from running.
+            # SEE https://github.com/ITISFoundation/osparc-simcore/issues/6318
+            _logger.warning(
+                "Skipping '%s:%s' due to %s",
+                service_key,
+                service_version,
+                err,
```
Comment (GitHK) [inline services/catalog/src/simcore_service_catalog/core/background_tasks.py:104]: Q: does putting the error type here help in some way?

**Review by GitHK — APPROVED** [review]
👍

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/catalog/services.py (hunk 2 above)
Comment (pcrespov) [reply packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/catalog/services.py:57]: I find that this way we have a good reference

File: services/catalog/src/simcore_service_catalog/core/background_tasks.py (hunk 3 above)
Comment (pcrespov) [reply services/catalog/src/simcore_service_catalog/core/background_tasks.py:104]: it shows the message in the error

**Pull request merged**